compiler/rustc_lint/messages.ftl

```diff
@@ -319,6 +319,8 @@ lint_invalid_reference_casting_borrow_as_mut = casting `&T` to `&mut T` is undef
 
 lint_invalid_reference_casting_note_book = for more information, visit <https://doc.rust-lang.org/book/ch15-05-interior-mutability.html>
 
+lint_invalid_reference_casting_note_ty_has_interior_mutability = `T` has interior mutability, but didn't go through `UnsafeCell::get` which is undefined behavior
```

Comment:
Who/what didn't go through `get`? This reads strangely to me right now.

Maybe say something a bit more general, like "even for types with interior mutability, the only legal way to obtain a mutable pointer from a shared reference is through `UnsafeCell::get`"?

Reply:
Done.